src/Core/Mews.Fiscalizations.Core/Compression/GzipCompressionHelper.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.IO.Compression;
+using System.Text;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace Mews.Fiscalizations.Core.Compression
+{
+    public static class GzipCompressionHelper
+    {
+        public static string Compress(string inputString)
+        {
+            byte[] buffer = Encoding.UTF8.GetBytes(inputString);
+            var memoryStream = new MemoryStream();
+            using (var gZipStream = new GZipStream(memoryStream, CompressionMode.Compress, true))
+            {
+                gZipStream.Write(buffer, 0, buffer.Length);
+            }
+
+            memoryStream.Position = 0;
+
+            var compressedData = new byte[memoryStream.Length];
+            memoryStream.Read(compressedData, 0, compressedData.Length);
+
+            var gZipBuffer = new byte[compressedData.Length + 4];
+            Buffer.BlockCopy(compressedData, 0, gZipBuffer, 4, compressedData.Length);
+            Buffer.BlockCopy(BitConverter.GetBytes(buffer.Length), 0, gZipBuffer, 0, 4);
+            return Convert.ToBase64String(gZipBuffer);
+        }
```

Comment:
This code is really hard to read and understand, also i believe it contains lots of weird logic

How about:

```
        public static byte[] Compress(string content, Encoding encoding)
        {
            var bytes = encoding.GetBytes(content);
            using var memoryStream = new MemoryStream();
            using (var gzipStream = new GZipStream(memoryStream, CompressionMode.Compress))
            {
                gzipStream.Write(bytes);
            }
            return memoryStream.ToArray();
        }
```

Reply:
one more thing, how about u make this a string extension? so you would do "hello world".Compress...etc